src/store/slices/selectedItemDetailsSlice.ts

```diff
@@ -0,0 +1,22 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { IBookDetails } from "../../interfaces";
+
+export interface SelectedItemDetailsState {
+  details: IBookDetails;
```

Comment:
```suggestion
  details: IBookDetails | null;
```
следующим комментом расписал